### ISSUE TITLE
MyRocks - "rocksdb-wal-size-limit-mb" seems to have no impact

### ISSUE THREAD

**Issue by lalitc18** [facebook/mysql-5.6#962]
If you check the reference to the variable rocksdb-max-total-wal-size and  rocksdb-wal-size-limit-mb  at the following address...

https://github.com/facebook/mysql-5.6/wiki/New-MySQL-RocksDB-Server-Variables

rocksdb-max-total-wal-size : Sets a limit on the maximum size of WAL files kept around. Once this limit is hit, RocksDB will force the flushing of memtables to reduce the size of WAL 

rocksdb-wal-size-limit-mb : Maximum size the RocksDB WAL is allow to grow to. When this size is exceeded rocksdb attempts to flush sufficient memtables to allow for the deletion of the oldest log.

Upon initial observation of the documentation, you would be lead to believe that these two variables do the same thing. This seemed a little strange so I decided to do some testing.

I started by setting rocksdb_max_total_wal_size to 16 Mb and rocksdb_wal_size_limit_mb to 32 Mb to see where the WAL file would be cut off for a memtable flush. As you'll see below by looking at WAL file 001700.log, it made it to 16 Mb before the WAL file was rotated. This is fine considering that if both variables do the same thing, we would see that cut off at 16 Mb. All is fine at this point.
```
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql   74 Jan 14 14:06 001700.log
[root@centos7-1 .rocksdb]# #started write
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql 565K Jan 14 14:07 001700.log
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql 2.1M Jan 14 14:07 001700.log
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql 3.1M Jan 14 14:07 001700.log
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql 4.7M Jan 14 14:07 001700.log
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql 6.6M Jan 14 14:07 001700.log
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql 8.2M Jan 14 14:07 001700.log
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql 9.9M Jan 14 14:07 001700.log
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:07 001700.log
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql  14M Jan 14 14:07 001700.log
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql  15M Jan 14 14:07 001700.log
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql  215 Jan 14 14:06 001692.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:06 001694.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:06 001699.sst
-rw-r----- 1 mysql mysql 339K Jan 14 14:07 001713.log
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:07 001715.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:07 001716.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:07 001717.sst
```
For my next test I swapped the two variables and set rocksdb_max_total_wal_size to 32 Mb and rocksdb_wal_size_limit_mb to 16 Mb. My assumption was that I was going to see a 16 Mb cut off again. However my test showed, as you can see below by look at file 001724.log, that the WAL file was cut off at 32 Mb

```
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:07 001716.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:07 001717.sst
-rw-r----- 1 mysql mysql  117 Jan 14 14:08 001718.log
-rw-r----- 1 mysql mysql 5.8M Jan 14 14:08 001720.sst
-rw-r----- 1 mysql mysql   74 Jan 14 14:08 001724.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:08 001730.sst
[root@centos7-1 .rocksdb]# #started write
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:07 001716.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:07 001717.sst
-rw-r----- 1 mysql mysql  117 Jan 14 14:08 001718.log
-rw-r----- 1 mysql mysql 5.8M Jan 14 14:08 001720.sst
-rw-r----- 1 mysql mysql 339K Jan 14 14:09 001724.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:08 001730.sst
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:07 001716.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:07 001717.sst
-rw-r----- 1 mysql mysql  117 Jan 14 14:08 001718.log
-rw-r----- 1 mysql mysql 5.8M Jan 14 14:08 001720.sst
-rw-r----- 1 mysql mysql 1.4M Jan 14 14:09 001724.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:08 001730.sst
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:07 001716.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:07 001717.sst
-rw-r----- 1 mysql mysql  117 Jan 14 14:08 001718.log
-rw-r----- 1 mysql mysql 5.8M Jan 14 14:08 001720.sst
-rw-r----- 1 mysql mysql 2.7M Jan 14 14:09 001724.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:08 001730.sst
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:07 001716.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:07 001717.sst
-rw-r----- 1 mysql mysql  117 Jan 14 14:08 001718.log
-rw-r----- 1 mysql mysql 5.8M Jan 14 14:08 001720.sst
-rw-r----- 1 mysql mysql 4.2M Jan 14 14:09 001724.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:08 001730.sst
......
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:07 001716.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:07 001717.sst
-rw-r----- 1 mysql mysql  117 Jan 14 14:08 001718.log
-rw-r----- 1 mysql mysql 5.8M Jan 14 14:08 001720.sst
-rw-r----- 1 mysql mysql  31M Jan 14 14:10 001724.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:08 001730.sst
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:07 001716.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:07 001717.sst
-rw-r----- 1 mysql mysql  117 Jan 14 14:08 001718.log
-rw-r----- 1 mysql mysql 5.8M Jan 14 14:08 001720.sst
-rw-r----- 1 mysql mysql  32M Jan 14 14:10 001724.log
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:08 001730.sst
[root@centos7-1 .rocksdb]# ls -lh | grep -e ".log" -e ".sst"
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001667.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001668.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:04 001670.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:06 001690.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:06 001691.sst
-rw-r----- 1 mysql mysql 6.0M Jan 14 14:06 001695.sst
-rw-r----- 1 mysql mysql 229K Jan 14 14:06 001696.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:07 001716.sst
-rw-r----- 1 mysql mysql 5.2M Jan 14 14:07 001717.sst
-rw-r----- 1 mysql mysql  117 Jan 14 14:08 001718.log
-rw-r----- 1 mysql mysql 5.8M Jan 14 14:08 001720.sst
-rw-r----- 1 mysql mysql 1.7K Jan 14 14:08 001730.sst
-rw-r----- 1 mysql mysql 792K Jan 14 14:10 001738.log
-rw-r----- 1 mysql mysql 1.2K Jan 14 14:10 001740.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:10 001741.sst
-rw-r----- 1 mysql mysql  12M Jan 14 14:10 001742.sst
-rw-r----- 1 mysql mysql  11M Jan 14 14:10 001743.sst
```
In conclusion I believe that, based on the documented definition, the "rocksdb-wal-size-limit-mb" variable is potentially deprecated and isn't being used by MyRocks. Either that or the documentation is incorrect.

**Comment by hermanlee** [comment]
Thanks for filing the issue. I think this is a problem with the documentation. According to the code comments:

```
  // The following two fields affect how archived logs will be deleted.
  // 1. If both set to 0, logs will be deleted asap and will not get into
  //    the archive.
  // 2. If WAL_ttl_seconds is 0 and WAL_size_limit_MB is not 0,
  //    WAL files will be checked every 10 min and if total size is greater
  //    then WAL_size_limit_MB, they will be deleted starting with the
  //    earliest until size_limit is met. All empty files will be deleted.
```

The options should control the size of the archived logged files, rather than the active WAL files which still back existing memtables. I'll update the documentation to reflect this.
